service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt

```diff
@@ -525,3 +525,25 @@ fun Database.Read.getReservationContractDayRanges(
         .mapTo<ChildContractDays>()
         .associate { it.childId to it.contractDays }
 }
+
+fun Database.Read.countReservationDays(childId: ChildId, range: FiniteDateRange): Int {
+    val sql =
+        """
+SELECT count(DISTINCT ar.date)
+FROM attendance_reservation ar
+JOIN realized_placement_one(ar.date) p ON p.child_id = ar.child_id
+JOIN daycare u ON u.id = p.unit_id
+WHERE ar.child_id = :childId
+AND ar.date BETWEEN :start AND :end
```

Comment:
IIRC `daterange(:start, :end, '[]') @> ar.date` or even `:range @> ar.date` would be more conventional

Reply:
`ar.date BETWEEN :start AND :end` hits indexes better, so this is not just about style.

Reply:
ah, good to know